lms/djangoapps/courseware/access.py

```diff
@@ -469,6 +469,10 @@ def _has_group_access(descriptor, user, course_key):
         # via updating the children of the split_test module.
         return ACCESS_GRANTED
 
+    # Allow staff group access, unless they are masquerading as a student.
+    if get_user_role(user, course_key) != 'student':
```

Comment:
As asked in another comment, is it possible to have roles other than Staff and Student? If so I think we should consider the change @andy-armstrong mentioned.